src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+import array
+import hightime
+import nirfsg
+import numpy as np
+import os
+import pathlib
+import pytest
+import sys
+import time
+
+
+sys.path.insert(0, str(pathlib.Path(__file__).parent.parent.parent / 'shared'))
+import system_test_utilities  # noqa: E402
+
+# Set up global information we need
+test_files_base_dir = os.path.join(os.path.dirname(__file__))
+use_simulated_session = True
+real_hw_resource_name = '5841'
+
+
+def get_test_file_path(file_name):
+    return os.path.join(test_files_base_dir, file_name)
+
+
+sys.path.insert(0, str(pathlib.Path(__file__).parent.parent.parent / 'generated/nirfsg'))
+
+
+class SystemTests:
+    @pytest.fixture(scope='function')
+    def test_rfsg_device_session(self, session_creation_kwargs):
+        if use_simulated_session:
+            with nirfsg.Session("5841sim", options="Simulate=1, DriverSetup=Model:5841", **session_creation_kwargs) as sim_5841_session:
+                yield sim_5841_session
+        else:
+            with nirfsg.Session(real_hw_resource_name, options="", **session_creation_kwargs) as real_rfsg_device_session:
+                yield real_rfsg_device_session
+
+# Attribute set and get related tests
+    def test_get_float_attribute(self, test_rfsg_device_session):
+        value = test_rfsg_device_session.power_level
+        assert isinstance(value, float)
+
+    def test_set_float_attribute(self, test_rfsg_device_session):
+        test_rfsg_device_session.power_level = -3.0
+        assert test_rfsg_device_session.power_level == -3.0
+
+    def test_get_string_attribute(self, test_rfsg_device_session):
+        model = test_rfsg_device_session.instrument_model
+        assert model == "NI PXIe-5841"
+
+    def test_set_string_attribute(self, test_rfsg_device_session):
+        test_rfsg_device_session.selected_script = "myScript"
+        assert test_rfsg_device_session.selected_script == "myScript"
+
+    def test_get_int32_attribute(self, test_rfsg_device_session):
+        value = test_rfsg_device_session.external_calibration_recommended_interval
+        assert isinstance(value, int)
+
+    def test_set_int32_enum_attribute(self, test_rfsg_device_session):
+        test_rfsg_device_session.frequency_settling_units = nirfsg.FrequencySettlingUnits.TIME_AFTER_LOCK
+        assert test_rfsg_device_session.frequency_settling_units == nirfsg.FrequencySettlingUnits.TIME_AFTER_LOCK
+
+    def test_set_invalid_attribute_raises(self, test_rfsg_device_session):
+        with pytest.raises(AttributeError):
+            test_rfsg_device_session.non_existent_attribute = 123
```

Comment:
```suggestion
```

You shouldn't need these. attribute get and set methods are covered by nifake unit tests, which we use to test commmon code.
I'm not sure we have anything in there to cover setting a non-existent attribute, but you're free to add such a test in a separate PR.